Apps/Src/ActivateMotor.c

```diff
@@ -0,0 +1,118 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+
+#include "ActivateMotor.h"
+
+
+static void motor_startup(OS_ERR *err) {
+    Precharge_Write(MOTOR_PRECHARGE, ON); // Activate the motor recharge
+
+    OSTimeDlyHMSM(0, 0, 5, 0, OS_OPT_TIME_HMSM_STRICT, &err);
+    // TODO: check for errors
+
+    Contactors_Set(MOTOR, ON); // Actually activate motor contactors
+    Precharge_Write(MOTOR_PRECHARGE, OFF);
+}
+
+
+void Task_MotorConnection(void *p_arg) {
+    car_state_t *car_state = (car_state_t *) p_arg;
+
+    OS_ERR err;
+    CPU_TS ts;
+
+    Contactors_Init(MOTOR);
+    motor_startup(&err);
+
+    // Now we need to spawn the four CAN related tasks
+
+    // Spawn SendCarCAN
+    OSTaskCreate(
+        (OS_TCB*)&SendCarCAN_TCB,
+        (CPU_CHAR*)"SendCarCAN",
+        (OS_TASK_PTR)Task_SendCarCAN,
+        (void*) p_arg,
+        (OS_PRIO)TASK_SEND_CAR_CAN_PRIO,
+        (CPU_STK*)SendCarCAN_Stk,
+        (CPU_STK_SIZE)WATERMARK_STACK_LIMIT,
+        (CPU_STK_SIZE)TASK_SEND_CAR_CAN_STACK_SIZE,
+        (OS_MSG_QTY)NULL,
+        (OS_TICK)NULL,
+        (void*)NULL,
+        (OS_OPT)(OS_OPT_TASK_STK_CLR),
+        (OS_ERR*)&err
+    );
+    // TODO: check for task creation error
+
+    // Spawn ReadCarCAN
+    OSTaskCreate(
```

Comment:
So is this task still being spawned by the ReadSwitches task, or is it just going to be using the Task Semaphore to block/unblock? If it's the latter, these task creations here need to only happen once it's determined that the motor is connected.

Reply:
It's still being spawned by ReadSwitches, but I think it makes sense to spawn it somewhere else now. The real question is whether we should spawn the four CAN tasks here or somewhere else, and have them also wait on a task semaphore.

Reply:
Up to you honestly. Just make sure for the tasks that should run freely once "spawned" that the semaphores aren't blocking on each iteration of the loop.